File: android/app/src/main/java/com/weighttracker/ui/screens/DietScreen.kt
package com.weighttracker.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.weighttracker.ui.viewmodel.MainViewModel
import java.text.SimpleDateFormat
import java.util.*

val MEAL_TYPES = listOf(
    Pair("breakfast", "早餐 🌅"),
    Pair("lunch", "午餐 ☀️"),
    Pair("dinner", "晚餐 🌙"),
    Pair("snack", "加餐 🍎")
)

@Composable
fun DietScreen(viewModel: MainViewModel = viewModel()) {
    val dietStats by viewModel.dietStats.collectAsState()
    val diets by viewModel.diets.collectAsState()
    
    var showAddDialog by remember { mutableStateOf(false) }
    var selectedMealType by remember { mutableStateOf(MEAL_TYPES[0]) }
    var foodName by remember { mutableStateOf("") }
    var calories by remember { mutableStateOf("") }
    var selectedDate by remember { mutableStateOf(SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(Date())) }
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Today Stats
        Card(
            modifier = Modifier.fillMaxWidth(),
            colors = CardDefaults.cardColors(
                containerColor = Color(0xFFF97316).copy(alpha = 0.1f)
            )
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Text("🥗 今日饮食", fontWeight = FontWeight.Bold, fontSize = 16.sp)
                Spacer(modifier = Modifier.height(12.dp))
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.Center
                ) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("${dietStats.todayCalories}", fontSize = 32.sp, fontWeight = FontWeight.Bold, color = Color(0xFFF97316))
                        Text("摄入 kcal", fontSize = 12.sp, color = Color.Gray)
                    }
                }
            }
        }
        
        // Add Diet Button
        Button(
            onClick = { showAddDialog = true },
            modifier = Modifier.fillMaxWidth()
        ) {
            Icon(Icons.Default.Add, contentDescription = null)
            Spacer(modifier = Modifier.width(8.dp))
            Text("记录饮食")
        }
        
        // Diet History
        Card(modifier = Modifier.fillMaxWidth()) {
            Column(modifier = Modifier.padding(16.dp)) {
                Text("📋 饮食历史", fontWeight = FontWeight.Bold)
                Spacer(modifier = Modifier.height(8.dp))
                
                if (diets.isEmpty()) {
                    Text("暂无饮食记录", color = Color.Gray, modifier = Modifier.padding(16.dp))
                } else {
                    diets.take(20).forEach { diet ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 8.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column {
                                Text(diet.mealName, fontWeight = FontWeight.Medium)
                                Text(diet.date, fontSize = 12.sp, color = Color.Gray)
                            }
                            
                            Text("${diet.calories} kcal", color = Color(0xFFF97316), fontWeight = FontWeight.Bold)
                        }
                    }
                }
            }
        }
    }
    
    // Add Diet Dialog
    if (showAddDialog) {
        AlertDialog(
            onDismissRequest = { showAddDialog = false },
            title = { Text("记录饮食") },
            text = {
                Column {
                    OutlinedTextField(
                        value = selectedDate,
                        onValueChange = { selectedDate = it },
                        label = { Text("日期") },
                        modifier = Modifier.fillMaxWidth()
                    )
                    
                    Spacer(modifier = Modifier.height(8.dp))
                    Text("餐次", fontSize = 12.sp, color = Color.Gray)
                    
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(4.dp)
                    ) {
                        MEAL_TYPES.forEach { (type, name) ->
                            FilterChip(
                                selected = selectedMealType.first == type,
                                onClick = { selectedMealType = Pair(type, name) },
                                label = { Text(name, fontSize = 10.sp) },
                                modifier = Modifier.weight(1f)
                            )
                        }
                    }
                    
                    Spacer(modifier = Modifier.height(8.dp))
                    OutlinedTextField(
                        value = foodName,
                        onValueChange = { foodName = it },
                        label = { Text("食物名称") },
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true
                    )
                    
                    Spacer(modifier = Modifier.height(8.dp))
                    OutlinedTextField(
                        value = calories,
                        onValueChange = { calories = it },
                        label = { Text("热量 (kcal)") },
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true
                    )
                }
            },
            confirmButton = {
                Button(
                    onClick = {
                        calories.toIntOrNull()?.let { cals ->
                            viewModel.addDiet(
                                date = selectedDate,
                                mealType = selectedMealType.first,
                                mealName = foodName.ifEmpty { selectedMealType.second },
                                mealIcon = "",
                                calories = cals
                            )
                            foodName = ""
                            calories = ""
                            showAddDialog = false
                        }
                    }
                ) {
                    Text("保存")
                }
            },
            dismissButton = {
                TextButton(onClick = { showAddDialog = false }) {
                    Text("取消")
                }
            }
        )
    }
}

File: flutter/lib/screens/diet_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:intl/intl.dart';
import '../providers/app_provider.dart';
import '../models/models.dart';

const _mealTypes = [
  ('breakfast', '早餐 🌅'),
  ('lunch', '午餐 ☀️'),
  ('dinner', '晚餐 🌙'),
  ('snack', '加餐 🍎'),
];

class DietScreen extends StatefulWidget {
  const DietScreen({super.key});

  @override
  State<DietScreen> createState() => _DietScreenState();
}

class _DietScreenState extends State<DietScreen> {
  String _selectedMealType = 'breakfast';
  final _foodController = TextEditingController();
  final _caloriesController = TextEditingController();
  late String _selectedDate;

  @override
  void initState() {
    super.initState();
    _selectedDate = DateFormat('yyyy-MM-dd').format(DateTime.now());
  }

  @override
  void dispose() {
    _foodController.dispose();
    _caloriesController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('🥗 饮食记录'),
        centerTitle: true,
      ),
      body: Consumer<AppProvider>(
        builder: (context, provider, child) {
          return SingleChildScrollView(
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                // Today Stats
                Card(
                  color: Colors.orange.withOpacity(0.1),
                  child: Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      children: [
                        const Text(
                          '🥗 今日饮食',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        const SizedBox(height: 12),
                        Text(
                          '${provider.todayDietCalories}',
                          style: const TextStyle(
                            fontSize: 36,
                            fontWeight: FontWeight.bold,
                            color: Colors.orange,
                          ),
                        ),
                        const Text('摄入 kcal'),
                      ],
                    ),
                  ),
                ),
                const SizedBox(height: 16),

                // Add Diet Button
                SizedBox(
                  width: double.infinity,
                  child: FilledButton.icon(
                    onPressed: () => _showAddDialog(context, provider),
                    icon: const Icon(Icons.add),
                    label: const Text('记录饮食'),
                  ),
                ),
                const SizedBox(height: 16),

                // History
                Card(
                  child: Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        const Text(
                          '📋 饮食历史',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        const SizedBox(height: 8),
                        if (provider.diets.isEmpty)
                          const Center(
                            child: Padding(
                              padding: EdgeInsets.all(16),
                              child: Text('暂无饮食记录'),
                            ),
                          )
                        else
                          ...provider.diets.reversed.take(20).map(
                            (diet) => ListTile(
                              contentPadding: EdgeInsets.zero,
                              leading: Text(
                                diet.mealIcon,
                                style: const TextStyle(fontSize: 24),
                              ),
                              title: Text(diet.mealName),
                              subtitle: Text(diet.date),
                              trailing: Text(
                                '${diet.calories} kcal',
                                style: const TextStyle(
                                  fontWeight: FontWeight.bold,
                                  color: Colors.orange,
                                ),
                              ),
                            ),
                          ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
          );
        },
      ),
    );
  }

  void _showAddDialog(BuildContext context, AppProvider provider) {
    _selectedMealType = 'breakfast';
    _foodController.clear();
    _caloriesController.clear();
    _selectedDate = DateFormat('yyyy-MM-dd').format(DateTime.now());

    showDialog(
      context: context,
      builder: (context) => StatefulBuilder(
        builder: (context, setState) {
          return AlertDialog(
            title: const Text('记录饮食'),
            content: SingleChildScrollView(
              child: Column(
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  TextField(
                    decoration: const InputDecoration(
                      labelText: '日期',
                      border: OutlineInputBorder(),
                    ),
                    controller: TextEditingController(text: _selectedDate),
                    onChanged: (v) => _selectedDate = v,
                  ),
                  const SizedBox(height: 12),
                  const Text('餐次'),
                  const SizedBox(height: 8),
                  Wrap(
                    spacing: 8,
                    children: _mealTypes.map((type) {
                      final isSelected = _selectedMealType == type.$1;
                      return FilterChip(
                        label: Text(type.$2),
                        selected: isSelected,
                        onSelected: (_) => setState(() => _selectedMealType = type.$1),
                      );
                    }).toList(),
                  ),
                  const SizedBox(height: 12),
                  TextField(
                    controller: _foodController,
                    decoration: const InputDecoration(
                      labelText: '食物名称',
                      border: OutlineInputBorder(),
                    ),
                  ),
                  const SizedBox(height: 12),
                  TextField(
                    controller: _caloriesController,
                    decoration: const InputDecoration(
                      labelText: '热量 (kcal)',
                      border: OutlineInputBorder(),
                    ),
                    keyboardType: TextInputType.number,
                  ),
                ],
              ),
            ),
            actions: [
              TextButton(
                onPressed: () => Navigator.pop(context),
                child: const Text('取消'),
              ),
              FilledButton(
                onPressed: () {
                  final calories = int.tryParse(_caloriesController.text) ?? 0;
                  if (calories > 0) {
                    final mealInfo = _mealTypes.firstWhere(
                      (t) => t.$1 == _selectedMealType,
                      orElse: () => _mealTypes[0],
                    );
                    provider.addDiet(DietRecord(
                      date: _selectedDate,
                      mealType: _selectedMealType,
                      mealName: _foodController.text.isEmpty
                          ? mealInfo.$2.split(' ').first
                          : _foodController.text,
                      mealIcon: mealInfo.$2.split(' ').last,
                      calories: calories,
                    ));
                    Navigator.pop(context);
                  }
                },
                child: const Text('保存'),
              ),
            ],
          );
        },
      ),
    );
  }
}
